cockatrice/src/tab_game.cpp

```diff
@@ -1814,10 +1819,16 @@ void TabGame::createMessageDock(bool bReplay)
         sayEdit->setCompleter(completer);
         actCompleterChanged();
 
-        if (spectator && !gameInfo.spectators_can_chat() && tabSupervisor->getAdminLocked()) {
+        /* Spectators can only talk if:
+         * (a) the game creator allows it
+         * (b) the spectator is a moderator/administrator
+         * (c) the spectator is a judge
+         */
+        if (!(isSpectator() && (gameInfo.spectators_can_chat() || !tabSupervisor->getAdminLocked() || isJudge()))) {
```

Comment:
is there a specific reason to use (and add) these public functions while they're simply available as private members of this object?

Reply:
Not really, I thought it was a bit cleaner. I can change that if you'd like